src/core/lib/security/security_connector/local/local_security_connector.cc

```diff
@@ -143,9 +143,13 @@ void local_check_peer(tsi_peer peer, grpc_endpoint* ep,
   }
   if (peer.properties != nullptr) gpr_free(peer.properties);
   peer.properties = new_properties;
-  // TODO(yihuazhang): Set security level of local TCP to TSI_SECURITY_NONE.
-  const char* security_level =
-      tsi_security_level_to_string(TSI_PRIVACY_AND_INTEGRITY);
+  // Set security level to NONE for TCP type, privacy&integrity otherwise.
+  const char* security_level;
+  if (type == LOCAL_TCP) {
```

Comment:
We need to guard the new functionality with a gRPC experiment, so we can safely roll it out internally. 

See https://github.com/grpc/grpc/pull/37035 as a recent example of how to create an experiment and use it in your code.